src/main/java/org/homework/step2/repository/TodoRepository.java

```diff
@@ -0,0 +1,41 @@
+package org.homework.step2.repository;
+
+import org.homework.step2.Todo;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Scanner;
+
+public class TodoRepository {
+    static private final Map<Integer, Todo> map = new HashMap<>();
+
+    private static int todoId = 0;
+
+    private static int getTodoId() {
+        return todoId++;
+    }
+
+    public Todo select(int id) {
+        return map.get(id);
+    }
+
+    public Todo insert(String description) {
+        Todo todo = new Todo(description, getTodoId());
+        map.put(todo.getId(), todo);
+        return todo;
+    }
+
+    public Todo delete(int id) {
+        return map.remove(id);
+    }
+
+    public Todo update(int id, String description) {
+        return Optional.ofNullable(map.get(id))
+                .map(todo -> {
+                    todo.setDescription(description);
+                    return todo;
+                })
+                .orElse(null);
+    }
+}
```

Comment:
```suggestion
public Todo update(int id, String description) {
    if (map.containsKey(id)) {
        Todo todo = map.get(id);
        todo.setDescription(description);
        return todo;
    }
    return null;
}

public Todo update(int id, String description) {
    Optional<Todo> todoOptional = Optional.ofNullable(map.get(id));
    todoOptional.ifPresent(todo -> todo.setDescription(description));
    return todoOptional.orElse(null);
}

```
Optional을 활용하는 모습 👍 
이건 못보고 넘어간거같네요~ 3단계에 Optional을 적용하는 요구사항이 있으니 3단계에서 적용해주시면 될거같습니다.

우선 더 정확한 사용법을 설명 드릴게요.

Optional을 사용할때 Null을 반환하게되면 Optional를 사용하는 이유가 없어집니다.
기본값을 반환하거나, 예외를 던지는게 가장 좋습니다. (위 케이스는 예외를 던지는게 맞겠죠?)

그리고 map은 주로 객체의 형태를 변환하는 데 사용되며, 객체의 상태를 변경하고싶다면 ifPresent와 같은 메소드를 활용하면 됩니다.